Excel.TemplateEngine/ObjectPrinting/Helpers/ChildSetterGenerator.cs

```diff
@@ -11,33 +10,24 @@
 
 namespace SkbKontur.Excel.TemplateEngine.ObjectPrinting.Helpers
 {
-    internal static class ObjectPropertySettersExtractor
+    internal static class ChildSetterGenerator
     {
         [NotNull]
-        public static Action<object> ExtractChildObjectSetter([NotNull] object model, [NotNull] ExcelTemplatePath path)
+        public static Expression<Action<object, object>> BuildChildSetterLambda([NotNull] Type parentType, [NotNull] ExcelTemplatePath childPath)
         {
-            var action = childObjectSettersCache.GetOrAdd((model.GetType(), path), x => ExtractChildModelSetter(x.type, x.path.PartsWithIndexers));
-            return x => action(model, x);
-        }
-
-        [NotNull]
-        private static Action<object, object> ExtractChildModelSetter([NotNull] Type modelType, [NotNull, ItemNotNull] string[] pathParts)
-        {
-            var targetModel = Expression.Variable(typeof(object));
-            var valueToSet = Expression.Parameter(typeof(object));
-            var currentModelNode = Expression.Convert(targetModel, modelType);
+            var parent = Expression.Parameter(typeof(object));
+            var child = Expression.Parameter(typeof(object));
+            var currentModelNode = Expression.Convert(parent, parentType);
 
-            var statements = BuildExtractionOfChildModelSetter(modelType, currentModelNode, valueToSet, pathParts);
+            var statements = BuildChildSetter(parentType, currentModelNode, child, childPath.PartsWithIndexers);
             var block = Expression.Block(new ParameterExpression[0], statements);
```

Comment:
у `Expression.Block` есть перегрузка без параметров
```suggestion
            var block = Expression.Block(statements);
```